### PR TITLE
fixes #5140 - User children taxonomies did not show tooltip 'Parent is already selected'

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
@eLobato, the users form is now using the `loc_org_tabs` partial which doesn't gray out the child taxonomies, but rather shows a tooltip 'Parent is already selected'. This is true for all 'taxable taxonomies' (i.e. users, domains, smart proxies, hostgroupsts, etc). If I remember correctly, we discussed the behavior and to add a tooltip rather than disabling child taxonomies and moving them to the selected column.  We can can discuss this behavior again.  I think it had to do with not submitting the id's of the child taxonomies and/or wanting to so that ensure certain child taxonomies remain selected if the parent taxonomy is removed.

**Comment by isratrade** [comment]
fyi, original issue #5140 is "children taxonomies are not greyed out"

**Comment by dLobatog** [comment]
I think it's okay to leave the tooltip rather than blocking. 

I see only one problem with this PR when I tested it, it only takes into account the status of the multi select when the page is loaded. 

If a parent taxonomy is moved to the left (unselected), it'll still show up the tooltip for children taxonomies. 


:+1: to merge after that. It's still a bit confusing but that's good enough I think.

**Comment by isratrade** [comment]
@eLobato, I discussed this with @abenari and we feel that the tool-tips are not so beneficial after a user starts deselecting items.  To keep the tool-tips in sync, it can't be work client-side with jquery and needs to make an ajax call which we think is overkill just for toolips.  For example, if there are multiple parents A and A/B that are selected and A/B/C is not selected.  When you de-select A/B, I can't automatically remove the tool-tip of it's descendants (C), since it's also a descendant of A. I need to make an ajax call to refresh
 `'data-descendants' => obj.children_of_selected_organization_ids.to_json`
Instead, I just remove the tooltip 'Parent is already selected' after an item is deselected.  It's a "hack" but I think the cost/benefit ratio justifies it.  What do others think?

**Comment by isratrade** [comment]
@thomasmckay, @ehelms, Jenkins is showing a failure on katello only, but it didn't show the failed test. Any ideas?

**Comment by domcleal** [comment]
@isratrade it's since been fixed

**Comment by isratrade** [comment]
[test]

**Comment by isratrade** [comment]
@eLobato, any changes needed here?

**Comment by dLobatog** [comment]
@isratrade I tried recreating what you described, but the tooltip does not get removed when any item is deselected. See below:

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<%= javascript 'users' %>
```
Comment (isratrade) [inline app/views/users/index.html.erb:1]: @eLobato, this is why the PR didn't work on the users form, but it works on the others. There was a silent javascript error that it didn't find `taxonomy_added`

**Comment by dLobatog** [comment]
Works well, I think the point for having https://github.com/theforeman/foreman/pull/1543 exists so that it can be merged for 1.5.x because this one will get in for 1.6, correct me if I'm wrong @domcleal ?

**Comment by domcleal** [comment]
@eLobato yup

**Comment by isratrade** [comment]
I agree. Let's merge PR #1543 so it can be cherry-picked for 1.5.x

**Comment by dLobatog** [comment]
Now that #1543 is in for 1.5.x, this goes in for 1.6.

Merged as https://github.com/theforeman/foreman/commit/18d7939e477781673614436b7e2fcfd461f8095f , thanks @isratrade

**Pull request closed without merging**